src/date-picker/test/position.test.ts

```diff
@@ -0,0 +1,106 @@
+import { expect, fixture, elementUpdated } from "@open-wc/testing";
+import "../index";
+
+describe("DatePicker", () => {
+  describe("position", () => {
+    it("should be dislayed calendar to left when right not enough width and long label", async () => {
+      const container = document.createElement("kuc-date-picker");
+      container.setAttribute("value", "2021-10-20");
+      container.setAttribute("label", "long label long label long label");
+
+      const divParentEl = document.createElement("div");
+      divParentEl.style.paddingLeft = window.innerWidth - 100 + "px";
+      divParentEl.appendChild(container);
+
+      const el = await fixture(divParentEl);
+      const inputEl = el.querySelector(
+        ".kuc-base-date__input"
+      ) as HTMLInputElement;
+
+      inputEl.click();
+      await elementUpdated(container);
+      await elementUpdated(el);
+      const calendaerEl = el.querySelector(
+        ".kuc-base-date__calendar"
+      ) as HTMLElement;
+      const computedStyle = window.getComputedStyle(calendaerEl);
+
+      expect(computedStyle.position).to.equal("absolute");
+      expect(computedStyle.top).to.equal("40px");
+    });
+
+    it("should be dislayed calendar to top when bottom not enough height", async () => {
```

Comment:
```suggestion
    it("should be displayed calendar to top when there is not enough height on the bottom", async () => {
```

And it will be easier to read this test case is placed on the very top or bottom. Now it is sandwiched with left/right test cases.